### DISCUSSION TITLE
Beginner Guide

### DISCUSSION THREAD

**Discussion by Iamcoded** [community/community#57342] [New to GitHub]
### Select Topic Area

Question

### Body

I really need to understand how this Github works, anyone have a quick beginner guide?

**Comment by NyanKaungSet** [comment]
Hello,

- [Hello World - GitHub Docs](https://docs.github.com/en/get-started/quickstart/hello-world)
- [GitHub w3schools](https://www.w3schools.com/git/git_remote_getstarted.asp)

Hope these can help you.

**Comment by Estrolldomezx** [comment]
It's an Internet hosting service for  version control your project codes.

**Comment by queenofcorgis** [comment]
Welcome to GitHub and the GitHub Community @Iamcoded  🎆 ! We're glad you're here and can't wait to see where your GitHub journey takes you. Check out this [check-in](https://github.com/orgs/community/discussions/42302#discussion-4683140) with links on how to choose your first project to contribute to, a blog with getting started resources, and a new user checklist.

Best of luck!

**Comment by DSDmark** [comment]
Hi @Iamcoded,
   
> Nice to meet you and welcome to this great community, we are here to help each other and try to make open source better. This is the discussions section if you want to discuss any. you can ask here, we will try to help.

And also, why any chance you are curious about, 

> what is GitHub, who is GitHub then I highly suggest you check this out [**What is GitHub?**](https://youtu.be/pBy1zgt0XPc).  its a video. so, just ***sit back and relax***

**Comment by Gustavonn07** [comment]
GitHub is a hosting and collaboration platform for software development projects using the Git version control system. It allows developers to work together, share, and collaborate on projects, as well as track and manage changes made to source code.

Here are the key concepts and functionalities of GitHub:

1. Repositories: A repository is a location where projects are stored. It can be public, accessible to everyone, or private, accessible only to project collaborators. Repositories are used to store files, source code, documentation, and any other project-related resources.

2. Version Control: GitHub uses Git, a distributed version control system, to track and manage changes made to a project's files. Version control enables you to maintain a history of all changes, review modifications, undo or fix errors, and collaborate with others on different versions of the project.

3. Branches: Branches are independent copies of a repository that allow you to work on changes without affecting the main version of the project. You can create new branches to add features, fix bugs, or experiment with new ideas. Once you have completed the changes, you can merge the branch with the main version.

4. Pull Requests: A pull request is a request to merge the changes made in a branch with the main version of the repository. It is a way to propose changes and initiate a discussion with project collaborators. Pull requests enable reviewing, commenting, and approving changes before they are incorporated into the main version.

5. Issues and Projects: GitHub provides features for tracking issues and managing projects. Issues are used to report bugs, request features, or discuss ideas related to the project. Projects allow you to organize and prioritize tasks, track progress, and assign tasks to collaborators.

These are just some of the functionalities of GitHub. The platform also offers team collaboration features such as wikis, discussions, continuous integration, automated deployment, and more. It is widely used by the developer community to host and collaborate on open-source and private software projects. You can use git itself to release your codes in repositories or download github desktop.

If you need any help there is the tutorial: https://docs.github.com/en/get-started/quickstart/hello-world.

**Reply by cpgatbyu** [reply]
I'm trying out the Hello World tutorial you mention.  That tutorial references a "'Propose Changes' box".  I see no "'Propose Changes' box".

**Reply by Gustavonn07** [reply]
it should appear like this for you to write what you changed in the code:

<hr>

![image](https://github.com/community/community/assets/84361085/9073f52b-814e-44bd-a4dd-fba491b77fb9)


<hr>

**Reply by cpgatbyu** [reply]
Why does the tutorial put "Propose Changes" in quotes when "Propose Changes" appears nowhere on your screenshot?

**Reply by Gustavonn07** [reply]
I really don't know, but the intention would be to explain the changes, so this should be the right space

**Reply by cpgatbyu** [reply]
I don't doubt it.  I just wonder why a tutorial(!) would use the wrong names.  To prepare newbies like myself for the hazards of collaborative editing?

**Reply by Gustavonn07** [reply]
I can't deny it's really confusing, I'm already used to github, it took me a while to reason what the tutorial was asking

**Comment by Saliya99** [comment]
Sure, here's a quick beginner guide to help you understand the basics of GitHub:

Sign Up and Create a Repository: The first step is to create a GitHub account. After that, you can create your first repository. A repository (or "repo") is like a project folder where all the related files and revision history are stored.

Understand the Basics: The main elements you'll work with in GitHub are repositories, branches, commits, and pull requests.

Repository: It's a project folder where all related files and revision history are stored.

Branch: It's a parallel version of a repository. It's contained within the repository, but does not affect the primary or main branch allowing you to work freely without disrupting the live or main version. When you've made the changes, you can pull request to merge your branch with the main branch.

Commit: It's a save changes command. Each commit saves a version of your project which you can revert to at any time.

Pull Request (PR): It's a proposal or request that your changes should be merged with the main branch. Other collaborators can review your changes and add comments before it gets merged.

Make and Commit Changes: Make some changes to your repository and then save these changes by committing them. You need to write a commit message which identifies your changes.

Open a Pull Request: After committing your changes, you open a pull request to merge your changes into the main branch.

Merge Pull Request: If your pull request is approved, you can merge your changes into the main branch. After merging, you can delete the branch where you have made the changes.

For more detailed information, you can check [Roger Dudler's GitHub Guide](https://rogerdudler.github.io/git-guide/), which provides a simple guide to get started with git and GitHub. The official [GitHub Guides](https://guides.github.com/) also provide a wealth of information.

Remember, practice makes perfect. Keep playing around and you'll get the hang of it!
